WPMedia/fusion-news-theme-blocks
# Tmedia 488 password validation two of

## Description
Sequential matching of password requirements 

## Jira Ticket
- [TMEDIA-488](https://arcpublishing.atlassian.net/browse/TMEDIA-488)

## Acceptance Criteria
<p data-pm-slice="1 1 []">The Mercury password rules are: </p>

Rules | Value
-- | --
Length | 12
Special Characters | 2
Lowercase | 2
Uppercase | 2
Numerical | 2

<p></p><p>The following passwords meet these rules but the signup form shows as errors:</p>

Password
<img width="711" alt="Screen Shot 2021-09-15 at 13 49 40" src="https://user-images.githubusercontent.com/5950956/133491970-2f719703-b06e-402e-b3fa-bea2dc3008c7.png">

## Test Steps

1. Use the latest version of fusion-news-theme with the-mercury
2. Checkout this branch `git checkout TMEDIA-488-password-validation-two-of`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/identity-block`
3. Go to http://localhost/pf/account/sign-up/?_website=the-mercury
4. Type in a password to fulfill the criteria with two uppercase for example not next to each other. Please see https://arcpublishing.atlassian.net/secure/RapidBoard.jspa?rapidView=875&modal=detail&selectedIssue=TMEDIA-488&assignee=5e387d6a1b1d910e5dfd7a9b and screenshot above for target strings 
5. Other sites with one requirement should still work as well and be able to submit and create new user 

## Effect Of Changes
### Before

- Non-sequential "aAAAAa" would not pass for lowercase check

### After
- non-sequential requirements work as expected 

## Dependencies or Side Effects


## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
